sphinx_gallery/tests/test_gen_gallery.py

```diff
@@ -345,7 +345,7 @@ def _check_order(sphinx_app, key, expected_order=None):
     content=_template_conf.format(
         imports="",
         subsection_order='["src/second-subsection", "src/first-subsection"]',
-        within_subsection_order='"NumberOfCodeLinesSortKey"',  # this is the default
+        within_subsection_order="sphinx_gallery.sorting.NumberOfCodeLinesSortKey",  # this is the default
```

Comment:
This was just because I realise we never test passing the `NumberOfCodeLinesSortKey` object, which we still support for backwards compatibility. I think this is okay as we test 'NumberOfCodeLinesSortKey' as string elsewhere, but let me know @drammock